Lib/_pydecimal.py

```diff
@@ -582,6 +582,21 @@ def __new__(cls, value="0", context=None):
 
         raise TypeError("Cannot convert %r to Decimal" % value)
 
+    @classmethod
+    def from_number(cls, number):
+        """Converts a real number to a decimal number, exactly.
+
+        >>> Decimal.from_number(314)              # int
+        Decimal('314')
+        >>> Decimal.from_number(0.1)              # float
+        Decimal('0.1000000000000000055511151231257827021181583404541015625')
+        >>> Decimal.from_number(Decimal('3.14'))  # another decimal instance
+        Decimal('3.14')
+        """
+        if isinstance(number, (int, Decimal, float)):
```

Comment:
Why this particular selection? We could add `Fraction` or `Rational` as well, which is perhaps more in line with https://github.com/python/cpython/issues/121797 and https://github.com/python/cpython/pull/121800

Reply:
Yes, these are the only numeric types accepted by the constructor. Exact conversion from Fraction to Decimal is not possible in general case (e.g. 1/3).

Reply:
But it's possible from the Integral type.

Reply:
The main constructor does not accept the Integral type.

`from_number()` only supports a subset of values for the main constructor.

Reply:
Ah, then this does make sense.